go-smb2 dependency panic when running smb templates

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/nuclei/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #nuclei channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version:
<!-- You can find current version of nuclei with "nuclei -version" -->
<!-- We only accept issues that are reproducible on the latest version of nuclei. -->
<!-- You can find the latest version of project at https://github.com/projectdiscovery/nuclei/releases/ -->
dev | main

### Current Behavior:
<!-- A concise description of what you're experiencing. -->
```console
 cmdutil ./nuclei -uc -ul 1000 -pt javascript -stats                                                              

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.1.8-dev

		projectdiscovery.io

[INF] Running uncover queries from template against: shodan
[INF] Current nuclei version: v3.1.8-dev (development)
[INF] Current nuclei-templates version: v9.7.4 (latest)
[WRN] Scan results upload to cloud is disabled.
[INF] New templates added in latest release: 6
[INF] Templates loaded for current scan: 23
[INF] Executing 23 signed templates from projectdiscovery/nuclei-templates
[INF] Targets loaded for current scan: 626
panic: runtime error: slice bounds out of range [:285216000] with capacity 338

goroutine 1864 [running]:
github.com/hirochachacha/go-smb2.(*conn).runReciever(0x1400cbed380)
	/Users/tarun/go/pkg/mod/github.com/hirochachacha/go-smb2@v1.1.0/conn.go:557 +0x6cc
created by github.com/hirochachacha/go-smb2.(*Negotiator).negotiate in goroutine 494
	/Users/tarun/go/pkg/mod/github.com/hirochachacha/go-smb2@v1.1.0/conn.go:102 +0x1e8

```

### Expected Behavior:
<!-- A concise description of what you expected to happen. -->
```console
 cmdutil ./nuclei -uc -ul 1000 -pt javascript -stats                                          

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.1.8-dev

		projectdiscovery.io

[INF] Running uncover queries from template against: shodan
[INF] Current nuclei version: v3.1.8-dev (development)
[INF] Current nuclei-templates version: v9.7.4 (latest)
[WRN] Scan results upload to cloud is disabled.
[INF] New templates added in latest release: 6
[INF] Templates loaded for current scan: 23
[INF] Executing 23 signed templates from projectdiscovery/nuclei-templates
[INF] Targets loaded for current scan: 626

--- a/cmd/integration-test/interactsh.go
+++ b/cmd/integration-test/interactsh.go
@@ -5,7 +5,7 @@ import osutils "github.com/projectdiscovery/utils/os"
 // All Interactsh related testcases
 var interactshTestCases = []TestCaseInfo{
 	{Path: "protocols/http/interactsh.yaml", TestCase: &httpInteractshRequest{}, DisableOn: func() bool { return osutils.IsWindows() || osutils.IsOSX() }},
-	{Path: "protocols/http/interactsh-stop-at-first-match.yaml", TestCase: &httpInteractshStopAtFirstMatchRequest{}, DisableOn: func() bool { return osutils.IsWindows() || osutils.IsOSX() }},
+	{Path: "protocols/http/interactsh-stop-at-first-match.yaml", TestCase: &httpInteractshStopAtFirstMatchRequest{}, DisableOn: func() bool { return true }},
 	{Path: "protocols/http/default-matcher-condition.yaml", TestCase: &httpDefaultMatcherCondition{}, DisableOn: func() bool { return true }}, // disable this test for now
 	{Path: "protocols/http/interactsh-requests-mc-and.yaml", TestCase: &httpInteractshRequestsWithMCAnd{}},
 }

--- a/go.mod
+++ b/go.mod
@@ -70,7 +70,6 @@ require (
 	github.com/go-pg/pg v8.0.7+incompatible
 	github.com/go-sql-driver/mysql v1.6.0
 	github.com/h2non/filetype v1.1.3
-	github.com/hirochachacha/go-smb2 v1.1.0
 	github.com/labstack/echo/v4 v4.10.2
 	github.com/lib/pq v1.10.1
 	github.com/mholt/archiver v3.1.1+incompatible
@@ -78,6 +77,7 @@ require (
 	github.com/praetorian-inc/fingerprintx v1.1.9
 	github.com/projectdiscovery/dsl v0.0.40
 	github.com/projectdiscovery/fasttemplate v0.0.2
+	github.com/projectdiscovery/go-smb2 v0.0.0-20240129200032-b4b580e6582b
 	github.com/projectdiscovery/goflags v0.1.35
 	github.com/projectdiscovery/gologger v1.1.12
 	github.com/projectdiscovery/gostruct v0.0.2

--- a/go.mod
+++ b/go.mod
@@ -77,7 +77,7 @@ require (
 	github.com/praetorian-inc/fingerprintx v1.1.9
 	github.com/projectdiscovery/dsl v0.0.40
 	github.com/projectdiscovery/fasttemplate v0.0.2
-	github.com/projectdiscovery/go-smb2 v0.0.0-20240129200032-b4b580e6582b
+	github.com/projectdiscovery/go-smb2 v0.0.0-20240129202741-052cc450c6cb
 	github.com/projectdiscovery/goflags v0.1.35
 	github.com/projectdiscovery/gologger v1.1.12
 	github.com/projectdiscovery/gostruct v0.0.2

--- a/pkg/js/compiler/compiler.go
+++ b/pkg/js/compiler/compiler.go
@@ -167,7 +167,12 @@ func (c *Compiler) ExecuteWithOptions(code string, args *ExecuteArgs, opts *Exec
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(opts.Timeout)*time.Second)
 	defer cancel()
 	// execute the script
-	results, err := contextutil.ExecFuncWithTwoReturns(ctx, func() (goja.Value, error) {
+	results, err := contextutil.ExecFuncWithTwoReturns(ctx, func() (val goja.Value, err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				err = errors.Errorf("panic: %v", r)
+			}
+		}()
 		return runtime.RunString(code)
 	})
 	if err != nil {

--- a/pkg/js/libs/smb/smb.go
+++ b/pkg/js/libs/smb/smb.go
@@ -5,8 +5,8 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/hirochachacha/go-smb2"
 	"github.com/praetorian-inc/fingerprintx/pkg/plugins"
+	"github.com/projectdiscovery/go-smb2"
 	"github.com/projectdiscovery/nuclei/v3/pkg/protocols/common/protocolstate"
 	"github.com/zmap/zgrab2/lib/smb/smb"
 )
@@ -14,7 +14,7 @@ import (
 // SMBClient is a client for SMB servers.
 //
 // Internally client uses github.com/zmap/zgrab2/lib/smb/smb driver.
-// github.com/hirochachacha/go-smb2 driver
+// github.com/projectdiscovery/go-smb2 driver
 type SMBClient struct{}
 
 // ConnectSMBInfoMode tries to connect to provided host and port

